plutus-ledger-api/src/PlutusLedgerApi/V1/Interval.hs

```diff
@@ -195,25 +239,38 @@ The resulting bound includes all values that are equal or smaller than the input
 upperBound :: a -> UpperBound a
 upperBound a = UpperBound (Finite a) True
 
-instance Ord a => JoinSemiLattice (Interval a) where
+-- See Note [Enumerable Intervals]
+instance (Enum a, Ord a) => JoinSemiLattice (Interval a) where
     {-# INLINABLE (\/) #-}
     (\/) = hull
 
-instance Ord a => BoundedJoinSemiLattice (Interval a) where
+-- See Note [Enumerable Intervals]
+instance (Enum a, Ord a) => BoundedJoinSemiLattice (Interval a) where
     {-# INLINABLE bottom #-}
     bottom = never
 
-instance Ord a => MeetSemiLattice (Interval a) where
+-- See Note [Enumerable Intervals]
+instance (Enum a, Ord a) => MeetSemiLattice (Interval a) where
     {-# INLINABLE (/\) #-}
     (/\) = intersection
 
-instance Ord a => BoundedMeetSemiLattice (Interval a) where
+-- See Note [Enumerable Intervals]
+instance (Enum a, Ord a) => BoundedMeetSemiLattice (Interval a) where
     {-# INLINABLE top #-}
     top = always
 
-instance Eq a => Eq (Interval a) where
+-- See Note [Enumerable Intervals]
+instance (Enum a, Ord a) => Eq (Interval a) where
     {-# INLINABLE (==) #-}
-    l == r = ivFrom l == ivFrom r && ivTo l == ivTo r
+    -- Degenerate case: both the intervals are empty.
+    -- There can be many empty intervals, so we check for this case
+    -- explicitly
+    iv1 == iv2 | isEmpty iv1 && isEmpty iv2 = True
```

Comment:
Why are empty Intervals equal? They do behave differently, for example if you call `hull` on them

Reply:
@michaelpj .